tmt/steps/provision/minute.py

```diff
@@ -111,39 +157,52 @@ def show(self):
 
     def wake(self, data=None):
         """ Override options and wake up the guest """
-        super().wake(['image', 'flavor', 'allow_ipv4_only'])
+        super().wake(['image', 'flavor', 'allow_ipv4_only', 'list', 'list-flavors'])
         if self.opt('dry'):
             return
 
         # Read API URL from 1minutetip script
-        try:
-            self.debug(f"Get the API URL from '{SCRIPT_PATH}'.")
-            script_content = self.read(SCRIPT_PATH)
-            match = re.search(API_URL_RE, script_content)
-            if not match:
-                raise tmt.utils.ProvisionError(
-                        f"Could not obtain API URL from '{SCRIPT_PATH}'.")
-            self.data['api_url'] = match.group('url')
-            self.debug('api_url', self.data['api_url'], level=3)
-        except tmt.utils.FileError:
-            raise tmt.utils.ProvisionError(
-                f"File '{SCRIPT_PATH}' not found. Please install 1minutetip.")
+        self._populate_api_url()
 
         if data:
+            data['api_url'] = self.api_url
             self._guest = GuestMinute(data, name=self.name, parent=self.step)
             self._guest.wake()
 
     def go(self):
-        """ Provision the container """
+        """ Provision the container, list flavors or list of images probided by openstack """
         super().go()
 
+        self._populate_api_url()
+
+        for opt in ['list', 'list-flavors']:
```

Comment:
Why is this in `for` loop? Imho it will be more readable as `if self.get("list"):` ...

Reply:
Don't know. It's just similar to code around.

Reply:
But you do not need loop. As I see there is no shared code between these options.

Reply:
Ok. Different solution has been provided.